servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -127,6 +127,12 @@ public void addTempBlockMeta(TempBlockMeta tempBlockMeta)
    */
   public void commitTempBlockMeta(TempBlockMeta tempBlockMeta)
       throws WorkerOutOfSpaceException, BlockAlreadyExistsException, BlockDoesNotExistException {
+    long blockId = tempBlockMeta.getBlockId();
+    if (hasBlockMeta(blockId)) {
```

Comment:
Is this necessary? `dir.addBlockMeta` on line 139 already throws `BlockAlreadyExistsException`.